### PR TITLE
add new API:paddle.moveaxis/Tensor.moveaxis

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

New API: `paddle.moveaxis/Tensor.moveaxis`
---

```
import paddle

x = paddle.randn([3, 5, 2, 4])
paddle.moveaxis(x, [0, 1], [2, 3]).shape
# [2, 4, 3, 5]

x = paddle.randn([2, 3])
paddle.moveaxis(x, 0, 1)       # equivalent to paddle.t(x)
# [3, 2]
```

infoflow 2021-12-14 14-45-48

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_transpose_op.py [hunk 1]
```diff
@@ -348,5 +348,77 @@ def test_x_dimension_check():
...
+        self.assertEqual(np.array_equal(out.numpy(), expected), True)
+        paddle.enable_static()
+
+    def test_error(self):
+        x = paddle.randn([2, 3, 4, 5])
+        # src must have the same number with dst
+        with self.assertRaises(AssertionError):
+            paddle.moveaxis(x, [1, 0], [2])
+
+        # each elment of src must be unique
```
Comment (zhhsplendid) [inline python/paddle/fluid/tests/unittests/test_transpose_op.py]: English typo? elment -> element.

Same for below.

Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_transpose_op.py]: All have been changed, thx

**Review by zhhsplendid — DISMISSED** [review]
LGTM

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -2580,3 +2581,103 @@ def as_real(x, name=None):
     outputs = {"Out": out}
     helper.append_op(type=op_type, inputs=inputs, outputs=outputs)
     return out
+
+
+def moveaxis(x, source, destination, name=None):
+    """
+    Move the ``source`` position of axis to ``destination`` position.
```
Comment (TCChenlong) [inline python/paddle/tensor/manipulation.py]: move the axis of tenso from source position to destination position
这样是不好一点？

Comment (zhwesky2010) [reply python/paddle/tensor/manipulation.py]: Done, thx

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**